plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -553,30 +574,162 @@ public void dropView(ConnectorSession session, SchemaTableName schemaViewName)
+        if (!replace && existing.isPresent()) {
+            if (ignoreExisting) {
+                return;
+            }
+            throw new TrinoException(ALREADY_EXISTS, "Materialized view already exists: " + schemaViewName);
+        }
+
+        if (existing.isPresent()) {
+            // drop the current storage table
+            String oldStorageTable = existing.get().getParameters().get(STORAGE_TABLE);
+            if (oldStorageTable != null) {
+                deleteTable(existing.get().getDatabaseName(), oldStorageTable);
+            }
+            // Replace the existing view definition
+            deleteTable(schemaViewName.getSchemaName(), schemaViewName.getTableName());
+        }
+        // Create the storage table
+        SchemaTableName storageTable = createMaterializedViewStorageTable(session, schemaViewName, definition);
+        // Create a view indicating the storage table
+        TableInput materializedViewTableInput = getMaterializedViewTableInput(
+                schemaViewName.getTableName(),
+                encodeMaterializedViewData(fromConnectorMaterializedViewDefinition(definition)),
+                session.getUser(),
+                createMaterializedViewProperties(session, storageTable.getTableName()));
+        createTable(schemaViewName.getSchemaName(), materializedViewTableInput);
     }
 
     @Override
     public void dropMaterializedView(ConnectorSession session, SchemaTableName schemaViewName)
     {
-        throw new TrinoException(NOT_SUPPORTED, "dropMaterializedView is not supported for Iceberg Glue catalogs");
+        com.amazonaws.services.glue.model.Table view = getTable(schemaViewName)
+                .orElseThrow(() -> new MaterializedViewNotFoundException(schemaViewName));
+
+        String storageTableName = view.getParameters().get(STORAGE_TABLE);
+        if (storageTableName != null) {
+            try {
+                dropTable(session, new SchemaTableName(schemaViewName.getSchemaName(), storageTableName));
+            }
+            catch (TrinoException e) {
+                LOG.warn(e, "Failed to drop storage table '%s' for materialized view '%s'", storageTableName, schemaViewName);
+            }
+        }
+        deleteTable(schemaViewName.getSchemaName(), schemaViewName.getTableName());
     }
 
     @Override
     protected Optional<ConnectorMaterializedViewDefinition> doGetMaterializedView(ConnectorSession session, SchemaTableName schemaViewName)
     {
-        return Optional.empty();
+        Optional<com.amazonaws.services.glue.model.Table> maybeTable = getTable(schemaViewName);
+        if (maybeTable.isEmpty()) {
+            return Optional.empty();
+        }
+
+        com.amazonaws.services.glue.model.Table table = maybeTable.get();
+        Map<String, String> tableParameters = table.getParameters();
+        if (!isPrestoView(tableParameters) || !isHiveOrPrestoView(table.getTableType()) || !tableParameters.containsKey(STORAGE_TABLE)) {
```

Comment:
Why don't we use `!(isPrestoView(table.getParameters()) && table.getParameters().get(TABLE_COMMENT).equalsIgnoreCase(ICEBERG_MATERIALIZED_VIEW_COMMENT))` like in `listMaterializedViews` ?
Maybe extract that into a `isTrinoMaterializedView` function

Reply:
Added `isTrinoMaterializedView`